src/lib/pages/frontsite/Documents/DocumentDetails/DocumentItems/DocumentItem.js

```diff
@@ -1,60 +1,111 @@
-import { getDisplayVal, invenioConfig } from '@config';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import Overridable from 'react-overridable';
-import { Grid, List } from 'semantic-ui-react';
+import { Button, Card, Table } from 'semantic-ui-react';
+import DocumentItemBody from './DocumentItemBody';
 
 class DocumentItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isShowingAll: false,
+      itemAmountLimit: 5,
+    };
+  }
+
+  get moreItemsToLoad() {
+    const { items } = this.props;
+    const { itemAmountLimit } = this.state;
+
+    return items.length > itemAmountLimit;
+  }
+
+  sliceItemsArray = (array) => {
+    const { itemAmountLimit } = this.state;
+
+    return array.slice(0, itemAmountLimit);
+  };
+
+  toggleItems = () => {
+    const { isShowingAll } = this.state;
+
+    this.setState({ isShowingAll: !isShowingAll });
+  };
+
   render() {
-    const { item } = this.props;
+    const { internalLocationName, items, showTitle } = this.props;
+    const { isShowingAll, itemAmountLimit } = this.state;
+
+    const previewArrayOfItems = this.sliceItemsArray(items);
```

Comment:
Very minor: given that you already have `itemAmountLimit` from the `state`, not sure that the extra function `sliceItemsArray` is really useful here.